File: samples/BOTH/async/Ping_Pong/jasync.c
#include <stdio.h>
int perank;
int count = 0;

jactivity_t *ping(int);

jactivity_t *regme(char *, jcallback);

void regcallback(char *msg) {
  if (msg != NULL)
    perank = atoi(msg);
  else
    perank = -1;

  printf("Perank %d\n", perank);

  ping(perank);
}

int counter = 1;

jasync pong() {
  //  usleep(50000);
    printf("pong received... %d\n", counter++);
    ping(perank);
}

int main() {
  printf("Registering...");
  regme("hello", regcallback);
  return 0;
}

File: samples/BOTH/async/Ping_Pong/jasync.js
var counter = 0;
var pecnt = 1;

jasync function ping(penum) {
    counter = counter + 1;
    console.log("ping received from PE:", penum, " count ", counter);
    pong();
}

jasync function regme(msg, cback) {
    
    console.log("registration received from ", msg);
    cback('' + pecnt);
    pecnt++;
}
